x-pack/plugins/reporting/server/routes/lib/jobs_query.ts

```diff
@@ -5,161 +5,162 @@
  */
 
 import { i18n } from '@kbn/i18n';
-import { errors as elasticsearchErrors } from 'elasticsearch';
+import {
+  CountParams,
+  CountResponse,
+  errors as elasticsearchErrors,
+  SearchParams,
+  SearchResponse,
+} from 'elasticsearch';
 import { get } from 'lodash';
 import { ReportingCore } from '../../';
-import { AuthenticatedUser } from '../../../../security/server';
-import { JobSource } from '../../types';
+import { JobSource, ReportingUser } from '../../types';
 
-const esErrors = elasticsearchErrors as Record<string, any>;
-const defaultSize = 10;
-
-interface QueryBody {
-  size?: number;
-  from?: number;
-  _source?: {
-    excludes: string[];
-  };
-  query: {
-    constant_score: {
-      filter: {
-        bool: {
-          must: Array<Record<string, any>>;
-        };
-      };
-    };
-  };
+// FIXME: typescript definition does not have the proper search API field name!
+declare module 'elasticsearch' {
+  interface SearchParams {
+    _source_excludes?: SearchParams['_sourceExclude'];
+  }
 }
 
-interface GetOpts {
-  includeContent?: boolean;
-}
+const esErrors = elasticsearchErrors as Record<string, any>;
+const defaultSize = 10;
 
-interface CountAggResult {
-  count: number;
-}
+type QueryType = 'search' | 'count';
 
-const getUsername = (user: AuthenticatedUser | null) => (user ? user.username : false);
+const getUsername = (user: ReportingUser) => (user ? user.username : false);
 
 export function jobsQueryFactory(reportingCore: ReportingCore) {
   const { elasticsearch } = reportingCore.getPluginSetupDeps();
   const { callAsInternalUser } = elasticsearch.legacy.client;
 
-  function execQuery(queryType: string, body: QueryBody) {
-    const defaultBody: Record<string, object> = {
-      search: {
-        _source: {
-          excludes: ['output.content'],
```

Comment:
Based on being guided by TS hints, I changed the way excludes works - it's `_source_excludes` on the search params object now.